src/main/java/org/gridsuite/modification/modifications/olg/OperationalLimitsGroupModification.java

```diff
@@ -519,46 +521,59 @@ private LoadingLimits.TemporaryLimit getTemporaryLimitToModify(
         return limitToModify;
     }
 
+    private boolean hasModification(AttributeModification<?> attribute) {
+        return attribute != null && attribute.getValue() != null && attribute.getOp().equals(OperationType.SET);
+    }
+
     public void modifyTemporaryLimit(
             CurrentLimitsAdder limitsAdder,
             CurrentTemporaryLimitModificationInfos limitModificationInfos,
             LoadingLimits.TemporaryLimit limitToModify,
             double limitValue,
             String limitDurationToReport,
-            String limitValueToReport,
             int limitAcceptableDuration,
             OperationalLimitsGroupInfos.Applicability applicability) {
-        if (Double.compare(limitToModify.getValue(), limitValue) != 0 && limitModificationInfos.getModificationType() != null) {
+        // Apply modifications only if specified, otherwise keep existing values
+        String finalName = hasModification(limitModificationInfos.getName())
+                ? limitModificationInfos.getName().getValue()
+                : limitToModify.getName();
+
+        double finalValue = hasModification(limitModificationInfos.getValue())
+                ? limitValue
+                : limitToModify.getValue();
+
+        // Check if there are any actual changes
+        boolean nameChanged = !limitToModify.getName().equals(finalName);
+        boolean valueChanged = Double.compare(limitToModify.getValue(), finalValue) != 0;
+
+        if (valueChanged && limitModificationInfos.getModificationType() != null) {
```

Comment:
Not your fault but I think it should be : 
```suggestion
        if (valueChanged && !nameChanged) {
```
because otherwise if the user changes name and value he only has the value change logged : 
<img width="611" height="213" alt="image" src="https://github.com/user-attachments/assets/4ceafc97-a906-4823-baa4-cd88544db2e5" />